app/models/session.rb

```diff
@@ -28,17 +38,122 @@ class Session < ApplicationRecord
 
   scope :active, -> { where(draft: false) }
 
+  delegate :name, to: :location
+
+  after_initialize :set_timeline_attributes
+  after_validation :set_timeline_timestamps
+
+  on_wizard_step :timeline, exact: true do
+    validates :consent_days_before,
+              presence: true,
+              inclusion: {
+                in: %w[default custom]
+              }
+    validates :consent_days_before_custom,
+              presence: true,
+              numericality: {
+                greater_than_or_equal_to: 10,
+                less_than_or_equal_to: 30
+              },
+              if: -> { consent_days_before == "custom" }
+
+    validates :reminder_days_after,
+              presence: true,
+              inclusion: {
+                in: %w[default custom]
+              }
+    validates :reminder_days_after_custom,
+              presence: true,
+              numericality: {
+                greater_than_or_equal_to: 2,
+                less_than_or_equal_to: 7
+              },
+              if: -> { reminder_days_after == "custom" }
+
+    validates :close_consent_on,
+              presence: true,
+              inclusion: {
+                in: %w[default custom]
+              }
+    validates :close_consent_at,
+              presence: true,
+              if: -> { close_consent_on == "custom" }
+  end
+
   def health_questions
     campaign.vaccines.first.health_questions
   end
 
-  delegate :name, to: :location
-
   def type
     campaign.name
   end
 
   def in_progress?
     date.to_date == Time.zone.today
   end
+
+  def form_steps
+    %i[timeline confirm]
+  end
+
+  def days_between_consent_and_session
+    (date - send_consent_at).to_i / 1.day
+  end
+
+  def days_between_consent_and_reminder
+    (send_reminders_at - send_consent_at).to_i / 1.day
+  end
+
+  private
+
+  def set_timeline_attributes
```

Comment:
It feels like this method and the one below are view / controller / UI concerns and don't really belong here. Maybe one day we should consider using a decorator or something similar.